xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-ui/src/main/resources/XWiki/Notifications/Code/NotificationsWatchUIX.xml

```diff
@@ -0,0 +1,656 @@
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>require(['jquery', 'xwiki-meta'], function ($, xm) {
+  'use strict';
+  
+  var handleSubmit = function () {
+    var chosenOption = $('#watchModal input[type="radio"]:checked').val();
+    // This should be improved as the REST endpoint might not be the same
+    var restUrl = xm.restURL;
+    var queryParams = new URLSearchParams({
+      "form_token": xm.form_token
+    });
+    var isFilterRemoval = false;
+    var method = "PUT";
+    if (chosenOption === 'watchPage') {
+      var url = restUrl;
+    } else if (chosenOption === 'watchSpace') {
+      var url = restUrl.replace(/\/pages.*/, "");
+    } else if (chosenOption === 'watchWiki') {
+      var url = restUrl.replace(/\/spaces.*/, "");
+    } else if (chosenOption == 'unwatchPage' || chosenOption == 'unblockPage') {
+      var url = restUrl;
+      method = 'DELETE';
+    } else if (chosenOption == 'unwatchSpace' || chosenOption == 'unblockSpace') {
+      var url = restUrl.replace(/\/pages.*/, "");
+      method = 'DELETE';
+    } else if (chosenOption == 'unwatchWiki' || chosenOption == 'unblockWiki') {
+      var url = restUrl.replace(/\/spaces.*/, "");
+      method = 'DELETE';
+    } else if (chosenOption == 'blockPage') {
+      var url = restUrl;
+      queryParams.append("ignore", "true");
+    } else if (chosenOption == 'blockSpace') {
+      var url = restUrl.replace(/\/pages.*/, "");
+      queryParams.append("ignore", "true");
+    } else if (chosenOption == 'blockWiki') {
+      var url = restUrl.replace(/\/spaces.*/, "");
+      queryParams.append("ignore", "true");
+    } else {
+      console.error("Option " + chosenOption + " is not yet supported!");
+    }
+    if (url) {
+      var ajaxUrl = url + "/notificationsWatches?" + queryParams.toString();
+      if (!isFilterRemoval) {
+        $.ajax(ajaxUrl, {
+          method: method
+        }).done(function () {
+          window.location.reload();
+        }).error(function (data) {
+          console.error("Error when processing the request: " + data);
+        });
+      }
+    }
+  };
+  
+  var initWatchModal = function () {
+    $('#watchModal .btn-primary').on('click', handleSubmit);
+    $('#watchModal input[type="radio"]').on('change', function () {
+      $('#watchModal .btn-primary').prop('disabled', false);
+    });
+    $('#watchModal input[type="radio"]').each(function () {
+      var radioButton = $(this);
+      var titleId = radioButton.attr('aria-labelledby');
+      $('#' + titleId).parent('a').on('click', function() {
+        radioButton.prop('checked', 'checked');
+        $('#watchModal .btn-primary').prop('disabled', false);
+      });
+    });
+  };
+  
+  (XWiki.isInitialized &amp;&amp; initWatchModal()) || document.observe('xwiki:dom:loaded',initWatchModal);
```

Comment:
New JavaScript code shouldn't use Prototype code (`document.observe` is a Prototype API).

Reply:
You're right, I most likely took back the examples from https://www.xwiki.org/xwiki/bin/view/Documentation/DevGuide/FrontendResources/JavaScriptAPI/#HDOMEvents28xwiki.js29 without thinking much about it: we should probably improve the doc there to remove the calls to Prototype.